src/main/java/com/exasol/adapter/properties/ValidatorFactory.java

```diff
@@ -89,15 +88,53 @@ public StringValidator matches(final String propertyName, final Pattern pattern)
      *
      * @param propertyName name of the property to validate
      * @param enumClass    enum to check against
+     * @param <T>          enum type
      * @return validator for checking that the given value is contained in the enum
-     * @param <T> enum type
      */
     public <T extends Enum<T>> PropertyValidator enumeration(final String propertyName,
             final Class<T> enumClass) {
         this.context.addKnownProperty(propertyName);
         return new EnumerationValidator<>(this.context, propertyName, enumClass);
     }
 
+    /**
+     * Creates a validator for validating a multi-select property.
+     * <p>
+     * Please note that due to type erasure in Java generics, we have logical duplication here between the type
+     * parameter and the class of the enum. This is unfortunately unavoidable. Empty values or values that are empty
+     * after splitting at the commas are invalid.
+     * </p>
+     * 
+     * @param propertyName name of the property to validate
+     * @param enumClass    enum defining the allowed values
+     * @param <T>          type of the enum defining the allowed values for the property
+     * 
+     * @return an instance of {@code MultiSelectValidator} for validating the property
+     */
+    public <T extends Enum<T>> PropertyValidator multiSelect(final String propertyName,
+            final Class<T> enumClass) {
+        return new MultiSelectValidator<>(this.context, propertyName, enumClass);
+    }
+
+    /**
+     * Creates a validator for validating a multi-select property.
+     * <p>
+     * Please note that due to type erasure in Java generics, we have logical duplication here between the type
+     * parameter and the class of the enum. This is unfortunately unavoidable.
+     * </p>
+     *
+     * @param propertyName name of the property to validate
+     * @param enumClass    enum defining the allowed values
+     * @param emptyAllowed {@code true} if empty values are allowed; {@code false} otherwise
+     * @param <T>          type of the enum defining the allowed values for the property
+     *
+     * @return an instance of {@code MultiSelectValidator} for validating the property
+     */
+    public <T extends Enum<T>> PropertyValidator multiSelect(final String propertyName, final Class<T> enumClass,
+            final boolean emptyAllowed) {
```

Comment:
```suggestion
    public <T extends Enum<T>> PropertyValidator multiSelectAllowEmpty(final String propertyName, final Class<T> enumClass) {
```
Avoid boolean args, use method name to make it explicit.